### PR TITLE
  Dconf: Add recipe

### PR DESCRIPTION
**Pull request by khallebal**

### CODE REVIEW

File: app-misc/dconf/dconf-0.38.0.recipe [hunk 1]
```diff
@@ -0,0 +1,96 @@
+SUMMARY="A simple key/value storage system optimised forreading"
+DESCRIPTION="dconf is a simple key/value \
```
Comment (diversys) [inline app-misc/dconf/dconf-0.38.0.recipe:2]: Please wrap at 80th character.

File: app-misc/dconf/dconf-0.38.0.recipe [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+storage system that is heavily optimised forreading. \
+This makes it an ideal system for storing user preferences \
+(which are read 1000s of times for each time the user changes one). \
+It was created with this usecase in mind."
+HOMEPAGE="https://homepage/of/projectx.org"
```
Comment (diversys) [inline app-misc/dconf/dconf-0.38.0.recipe:7]: Seems missing.

File: app-misc/dconf/dconf-0.38.0.recipe [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+COPYRIGHT="2014 Developer name
```
Comment (diversys) [inline app-misc/dconf/dconf-0.38.0.recipe:8]: Ditto.

**Review by diversys — CHANGES_REQUESTED** [review]

File: app-misc/dconf/dconf-0.38.0.recipe (hunk 1 above)
Comment (Begasus) [reply app-misc/dconf/dconf-0.38.0.recipe:2]: This is now 100th character ;)

**Comment by khallebal** [comment]
Fixed, Please check again.

File: app-misc/dconf/dconf-0.38.0.recipe [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+It was created with this usecase in mind."
+HOMEPAGE="https://gitlab.gnome.org/"
+COPYRIGHT="2010 Ryan Lortie
+	2010 Tomasz Miąsko"
+LICENSE="
+	GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://gitlab.gnome.org/GNOME/dconf/-/archive/$portVersion/dconf-$portVersion.zip"
+CHECKSUM_SHA256="7dbcc81ef9d7ad043da2ac580ab651b9e572217a81175babff811880186879f1"
+SOURCE_DIR="$portVersionedName"
```
Comment (Begasus) [inline app-misc/dconf/dconf-0.38.0.recipe:17]: This doesn't work for secondary architecture on 32bit:
```diff
- SOURCE_DIR="$portVersionedName"
+ SOURCE_DIR="dconf-$portVersion"
```

File: app-misc/dconf/dconf-0.38.0.recipe [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+libdconfVersion=1.0.0
```
Comment (Begasus) [inline app-misc/dconf/dconf-0.38.0.recipe:22]: Use libVersionCompat here (and in PROVIDES + PROVIDES_devel)

**Comment by Begasus** [comment]
meson is broken atm after an update, trying a build with previous version there

**Comment by Begasus** [comment]
You can add a testcase here too:
```
TEST()
{
	ninja test -C build
}

File: app-misc/dconf/dconf-0.40.0.recipe [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://gitlab.gnome.org/GNOME/dconf/-/archive/$portVersion/dconf-$portVersion.zip"
+CHECKSUM_SHA256="f83e7e993ddca6d0e29ffa659a8b0acce80093cae0e24a60405e5427dbb4009b"
+SOURCE_DIR="$portVersionedName"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="?x86"
+
+libVersionCompat=1.0.0
```
Comment (Begasus) [inline app-misc/dconf/dconf-0.40.0.recipe:19]: Check this for libVersion/libVersionCompat: https://github.com/haikuports/haikuports/blob/4f180bdb94db9749ae800820b271e262a12754aa/media-libs/openexr/openexr-2.2.1.recipe#L22

File: app-misc/dconf/dconf-0.40.0.recipe [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+(which are read 1000s of times for each time the user changes one). \
+It was created with this usecase in mind."
+HOMEPAGE="https://gitlab.gnome.org/GNOME/dconf/"
+COPYRIGHT="2010 Ryan Lortie
+	2010 Tomasz Miąsko"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://gitlab.gnome.org/GNOME/dconf/-/archive/$portVersion/dconf-$portVersion.zip"
+CHECKSUM_SHA256="f83e7e993ddca6d0e29ffa659a8b0acce80093cae0e24a60405e5427dbb4009b"
+SOURCE_DIR="$portVersionedName"
```
[the remark this answers is not in the document]
Comment (Begasus) [reply app-misc/dconf/dconf-0.40.0.recipe:14]: Actualy this isn't needed at all (checked 32bit just now):
```diff
- SOURCE_DIR="$portVersionedName"
 ```

File: app-misc/dconf/dconf-0.40.0.recipe [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="?x86"
```
Comment (Begasus) [inline app-misc/dconf/dconf-0.40.0.recipe:17]: Why disable x86 here?

**Comment by Begasus** [comment]
As we follow Gentoo's layout, this package should go into gnome-base: https://packages.gentoo.org/packages/gnome-base/dconf

File: app-misc/dconf/dconf-0.40.0.recipe [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+		--bindir=$binDir \
+		--datadir=$dataDir \
+		--includedir=$includeDir \
+		--libdir=$libDir \
+		--libexecdir=$libDir \
+		--localedir=$dataDir/locale \
+		--mandir=$manDir
+
+	ninja -C build $jobArgs
+
```
Comment (Begasus) [inline app-misc/dconf/dconf-0.40.0.recipe:80]: No empty line here:
```diff
-
```

File: app-misc/dconf/dconf-0.40.0.recipe [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+INSTALL()
+{
+	ninja -C build install
+
+	prepareInstalledDevelLibs libdconf
```
Comment (Begasus) [inline app-misc/dconf/dconf-0.40.0.recipe:87]: Only one library so:
```diff
- 	prepareInstalledDevelLibs libdconf
+ 	prepareInstalledDevelLib libdconf
```

File: app-misc/dconf/dconf-0.40.0.recipe [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+	fixPkgconfig
+
+	packageEntries devel \
+		$developDir
+
```
Comment (Begasus) [inline app-misc/dconf/dconf-0.40.0.recipe:92]: No empty line here:
```diff
-
```

**Review by Begasus — CHANGES_REQUESTED** [review]

File: app-misc/dconf/dconf-0.40.0.recipe (hunk 11 above)
Comment (khallebal) [reply app-misc/dconf/dconf-0.40.0.recipe]: I've fixed the empty lines, please check again.

**Comment by Begasus** [comment]
LGTM, DESCRIPTION lines could be made longer (up to 100 characters) but is not a deal breaker here, thanks!

**Pull request merged**